### PR TITLE
perf - log slow running workbench contribs

### PR DESCRIPTION
**Pull request by bpasero**

### CODE REVIEW

File: src/vs/workbench/common/contributions.ts [hunk 1]
```diff
@@ -126,11 +128,20 @@ class WorkbenchContributionsRegistry implements IWorkbenchContributionsRegistry
...
 			instantiationService.createInstance(ctor);
 		} catch (error) {
-			console.error(`Unable to instantiate workbench contribution ${ctor.name}.`, error);
+			logService.error(`Unable to instantiate workbench contribution ${ctor.name}.`, error);
+		}
+
+		if (typeof now === 'number') {
+			const time = Date.now() - now;
+			if (time > 5) {
+				logService.warn(`Workbench contribution ${ctor.name} blocked restore phase by ${time}ms.`);
```
Comment (jrieken) [inline src/vs/workbench/common/contributions.ts:143]: Careful - in the product (after minification) this is just `Ke` or `Xa` etc

**Review by jrieken — DISMISSED** [review]

File: src/vs/workbench/common/contributions.ts (hunk 1 above)
Comment (bpasero) [reply src/vs/workbench/common/contributions.ts:143]: @jrieken ouch, any advice what to log then? I guess we would have to introduce identifiers to get out of it...

Comment (jrieken) [reply src/vs/workbench/common/contributions.ts:143]: Not really. You can just print the source of the ctor. I do something similar with extension language features - I know the extension but if its suggest or definitions etc. I then just print the caller: https://github.com/microsoft/vscode/blob/342394d1e7d43d3324dc2ede1d634cffd52ba159/src/vs/workbench/api/common/extHostLanguageFeatures.ts#L1891

Comment (bpasero) [reply src/vs/workbench/common/contributions.ts:143]: Ok let me see to add an identifier then.

Comment (bpasero) [reply src/vs/workbench/common/contributions.ts]: @jrieken can you approve again, `registerWorkbenchContribution` now carries an `id` that we can use to print it.

**Review by jrieken — DISMISSED** [review]
poor bundle size...

**Comment by bpasero** [comment]
Yeah indeed sucks, I don't mind unwinding this when we investigated the offenders...

**Review by joaomoreno — APPROVED** [review]

**Pull request merged**